### DISCUSSION TITLE
v1.121.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#14257] [Announcements]
# v1.121.0

## Highlights

Welcome to release `v1.121.0` of Immich. This release focuses on bug fixes and performance optimization across the app. Some of the highlights below:

* `@immich/ui` component library
* Fallback to system fonts for Cyrillic letters on the mobile app
* Multiselect using the shift key in the search result view
* **Notable fixes**: album sync on the mobile app always ran when reopening the app, leading to degradation in performance and browsing 

### `@immich/ui` component library

Some of you may have seen that we added a [new repository](https://github.com/immich-app/ui) on GitHub. We have started building out a shared UI components library for Immich that will hopefully lead to a more manageable and consistent design. It’s still very much in the works, but here is a sneak peek at the button component.

![image](https://github.com/user-attachments/assets/e2baebe9-9f55-42d5-8b7b-dca9872cb2f1)

### Notable fix: album sync degrades performance on the mobile app

We fixed a bug on the server that caused visible performance degradation of the mobile app. The album syncing process is always triggered when the app is reopened instead of only syncing when there are changes on the server. This is a server fix, so we encourage you to update your server to this new release to fix this performance issue.

## Support Immich

<p align="center">

<img src="https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExbjY2eWc5Y2F0ZW56MmR4aWE0dDhzZXlidXRmYWZyajl1bWZidXZpcyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/87CKDqErVfMqY/giphy.gif" width="450" title="SUPPORT THE PROJECT!"> 

</p>

If you find the project helpful, you can support Immich by purchasing a product key at https://buy.immich.app/.

Cheers! 🍻

### 🚀 Features

* feat: add minimal devcontainer setup by @mcarbonne in <https://github.com/immich-app/immich/pull/14038>

### 🌟 Enhancements

* refactor(mobile): video controls by @mertalev in <https://github.com/immich-app/immich/pull/14086>
* feat: use dateTimeOriginal to calculate album date by @p2kmgcl in <https://github.com/immich-app/immich/pull/14119>
* feat(mobile): new video slider ui by @mertalev in <https://github.com/immich-app/immich/pull/14126>
* chore(web): migration svelte 5 syntax by @alextran1502 in <https://github.com/immich-app/immich/pull/13883>
* feat(web): Added tag button to the context menu in the favorites page by @IMBeniamin in <https://github.com/immich-app/immich/pull/14156>
* feat: adding photo & video storage space to server stats by @weathondev in <https://github.com/immich-app/immich/pull/14125>
* feat(mobile): exclude locales from overpass font by @johnstef99 in <https://github.com/immich-app/immich/pull/14158>
* feat(web): Implement keep this delete others for asset stacks  by @bdavis2-PCTY in <https://github.com/immich-app/immich/pull/14217>
* feat: Added shortcuts, shift-multi select, and missing menu options to Search (Galleryviewer) by @weathondev in <https://github.com/immich-app/immich/pull/14213>
* feat(server): faster geodata import by @mertalev in <https://github.com/immich-app/immich/pull/14241>

### 🐛 Bug fixes

* fix(mobile): fix logout timeout by @johnstef99 in <https://github.com/immich-app/immich/pull/14104>
* fix(server): Some MTS videos fail to generate thumbnail by @Lukasdotcom in <https://github.com/immich-app/immich/pull/14134>
* fix(web): textarea autogrow height by @duckimann in <https://github.com/immich-app/immich/pull/13983>
* fix: Routing back button in sharedLinks page by @Pranay-Pandey in <https://github.com/immich-app/immich/pull/13703>
* fix(web): ensure current asset index stays within bounds by @michelheusschen in <https://github.com/immich-app/immich/pull/14013>
* fix(web): saving pasted coordinates by @michelheusschen in <https://github.com/immich-app/immich/pull/14143>
* fix(web): update description height when navigating between assets by @michelheusschen in <https://github.com/immich-app/immich/pull/14145>
* fix(web): allow selecting people after clearing search options by @michelheusschen in <https://github.com/immich-app/immich/pull/14146>
* fix(mobile): Android local notification failed to invoke by @alextran1502 in <https://github.com/immich-app/immich/pull/14155>
* fix(web): prevent infinite loop when modifying stacked asset by @michelheusschen in <https://github.com/immich-app/immich/pull/14162>
* fix(cli): Concurrency not fully using queue potential by @Tiefseetauchner in <https://github.com/immich-app/immich/pull/11828>
* fix(web): don't refresh the panorama viewer when modifying asset by @michelheusschen in <https://github.com/immich-app/immich/pull/14163>
* fix(mobile): unnecessary rebuilds from partner share notifier by @mertalev in <https://github.com/immich-app/immich/pull/14170>
* fix: show tags when viewing stacked assets by @michelheusschen in <https://github.com/immich-app/immich/pull/14199>
* fix(mobile): use sets in album refresh, concurrent futures by @mertalev in <https://github.com/immich-app/immich/pull/14193>
* fix(mobile): Dismissible menus by @johnstef99 in <https://github.com/immich-app/immich/pull/14192>
* fix(web): layout shifting when scrolling up by @alextran1502 in <https://github.com/immich-app/immich/pull/14226>
* fix(mobile): make search page scrollable by @johnstef99 in <https://github.com/immich-app/immich/pull/14228>
* fix(mobile): fixes on language change by @johnstef99 in <https://github.com/immich-app/immich/pull/14089>
* fix: mobile album sync is always triggered when opening the app by @alextran1502 in <https://github.com/immich-app/immich/pull/14233>
* fix(server): remove unnecessary guc settings for vector search by @mertalev in <https://github.com/immich-app/immich/pull/14237>
* fix: parse quota claim as a number by @danieldietzler in <https://github.com/immich-app/immich/pull/14178>

### 📚 Documentation

* chore(docs): Encode db dump in UTF-8 without BOM for Windows by @amorphobia in <https://github.com/immich-app/immich/pull/13775>
* docs: Update TrueNAS docs for TrueNAS SCALE 24.10 by @NicholasFlamy in <https://github.com/immich-app/immich/pull/14067>
* docs: get asset owner by ID from the database by @mmomjian in <https://github.com/immich-app/immich/pull/14174>
* docs: backup only selected photos by @**[shivanshs9](https://github.com/shivanshs9) in <https://github.com/immich-app/immich/pull/14225>**

## New Contributors

* @amorphobia made their first contribution in <https://github.com/immich-app/immich/pull/13775>
* @p2kmgcl made their first contribution in <https://github.com/immich-app/immich/pull/14119>
* @johnstef99 made their first contribution in <https://github.com/immich-app/immich/pull/14104>
* @duckimann made their first contribution in <https://github.com/immich-app/immich/pull/13983>
* @IMBeniamin made their first contribution in <https://github.com/immich-app/immich/pull/14156>
* @weathondev made their first contribution in <https://github.com/immich-app/immich/pull/14125>
* @bdavis2-PCTY made their first contribution in <https://github.com/immich-app/immich/pull/14217>

**Full Changelog**: <https://github.com/immich-app/immich/compare/v1.120.2...v1.121.0>

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.121.0'>v1.121.0</a>.</em>

**Comment by HELPY4** [comment]
Optimization 📈

**Comment by yu-jingrui** [comment]
Great update. This one fixed a lot of bugs for me. The Android app is finally usable.

**Comment by ufizo** [comment]
My server crashes with this update:

Rolling back to 1.120.2 works.

````immich_server            | [Nest] 7  - 11/24/2024, 4:22:26 PM     LOG [Microservices:EventRepository] Initialized websocket server
immich_server            | Migration "NaturalEarthCountriesIdentityColumn1732072134943" failed, error: type "public.earth" does not exist
immich_server            | QueryFailedError: type "public.earth" does not exist
immich_server            |     at PostgresQueryRunner.query (/usr/src/app/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:219:19)
immich_server            |     at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
immich_server            |     at async NaturalEarthCountriesIdentityColumn1732072134943.up (/usr/src/app/dist/migrations/1732072134943-NaturalEarthCountriesIdentityColumn.js:9:9)
immich_server            |     at async MigrationExecutor.executePendingMigrations (/usr/src/app/node_modules/typeorm/migration/MigrationExecutor.js:225:17)
immich_server            |     at async DataSource.runMigrations (/usr/src/app/node_modules/typeorm/data-source/DataSource.js:265:35)
immich_server            |     at async DatabaseRepository.runMigrations (/usr/src/app/dist/repositories/database.repository.js:189:9)
immich_server            |     at async /usr/src/app/dist/services/database.service.js:96:17
immich_server            |     at async /usr/src/app/dist/repositories/database.repository.js:197:23 {
immich_server            |   query: '\n' +
immich_server            |     '      CREATE FUNCTION ll_to_earth_public(latitude double precision, longitude double precision) RETURNS public.earth PARALLEL SAFE IMMUTABLE STRICT LANGUAGE SQL AS $$\n' +
immich_server            |     '        SELECT public.cube(public.cube(public.cube(public.earth()*cos(radians(latitude))*cos(radians(longitude))),public.earth()*cos(radians(latitude))*sin(radians(longitude))),public.earth()*sin(radians(latitude)))::public.earth\n' +
immich_server            |     '    $$',
immich_server            |   parameters: undefined,
immich_server            |   driverError: error: type "public.earth" does not exist
immich_server            |       at /usr/src/app/node_modules/pg/lib/client.js:535:17
immich_server            |       at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
immich_server            |       at async PostgresQueryRunner.query (/usr/src/app/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:184:25)
immich_server            |       at async NaturalEarthCountriesIdentityColumn1732072134943.up (/usr/src/app/dist/migrations/1732072134943-NaturalEarthCountriesIdentityColumn.js:9:9)immich_server            |       at async MigrationExecutor.executePendingMigrations (/usr/src/app/node_modules/typeorm/migration/MigrationExecutor.js:225:17)
immich_server            |       at async DataSource.runMigrations (/usr/src/app/node_modules/typeorm/data-source/DataSource.js:265:35)
immich_server            |       at async DatabaseRepository.runMigrations (/usr/src/app/dist/repositories/database.repository.js:189:9)
immich_server            |       at async /usr/src/app/dist/services/database.service.js:96:17
immich_server            |       at async /usr/src/app/dist/repositories/database.repository.js:197:23 {
immich_server            |     length: 104,
immich_server            |     severity: 'ERROR',
immich_server            |     code: '42704',
immich_server            |     detail: undefined,
immich_server            |     hint: undefined,
immich_server            |     position: undefined,
immich_server            |     internalPosition: undefined,
immich_server            |     internalQuery: undefined,
immich_server            |     where: undefined,
immich_server            |     schema: undefined,
immich_server            |     table: undefined,
immich_server            |     column: undefined,
immich_server            |     dataType: undefined,
immich_server            |     constraint: undefined,
immich_server            |     file: 'functioncmds.c',
immich_server            |     line: '136',
immich_server            |     routine: 'compute_return_type'
immich_server            |   },
immich_server            |   length: 104,
immich_server            |   severity: 'ERROR',
immich_server            |   code: '42704',
immich_server            |   detail: undefined,
immich_server            |   hint: undefined,
immich_server            |   position: undefined,
immich_server            |   internalPosition: undefined,
immich_server            |   internalQuery: undefined,
immich_server            |   where: undefined,
immich_server            |   schema: undefined,
immich_server            |   table: undefined,
immich_server            |   column: undefined,
immich_server            |   dataType: undefined,
immich_server            |   constraint: undefined,
immich_server            |   file: 'functioncmds.c',
immich_server            |   line: '136',
immich_server            |   routine: 'compute_return_type'
immich_server            | }
immich_server            | microservices worker error: QueryFailedError: type "public.earth" does not exist
immich_server            | microservices worker exited with code 1
immich_server            | Killing api process
immich_server exited with code 0
```

**Reply by alextran1502** [reply]
Please see this comment to fix your instance 

https://github.com/immich-app/immich/discussions/14269#discussioncomment-11347185
